### PR TITLE
prov/efa: Fix and clean up the handshake procedures in msg/rma

### PR DESCRIPTION
**Pull request by shijin-aws**
This PR contains two commits to fix and clean up the handshake procedures in msg and rma operations

### CODE REVIEW

File: prov/efa/src/rdm/efa_rdm_rma.c [hunk 1]
```diff
@@ -197,24 +197,25 @@ ssize_t efa_rdm_rma_readmsg(struct fid_ep *ep, const struct fi_msg_rma *msg, uin
...
+	}
+
 	use_lower_ep_read = false;
 	if (efa_both_support_rdma_read(efa_rdm_ep, peer)) {
 		/* efa_both_support_rdma_read also check domain.use_device_rdma,
 		 * so we do not check it here
 		 */
 		use_lower_ep_read = true;
 	} else if (efa_mr_is_neuron(txe->desc[0])) {
-		err = efa_rdm_ep_determine_rdma_read_support(efa_rdm_ep, txe->addr, peer);
```
Comment (shijin-aws) [inline prov/efa/src/rdm/efa_rdm_rma.c:207]: Personally I think we should just remove this `else if`. No code is reaching this code path AFAICT.

**Review by wzamazon — APPROVED** [review]
I think the variable `use_lower_ep_read` should be renamed to `use_device_read`

**Review by wenduwan — APPROVED** [review]

**Comment by shijin-aws** [comment]
> I think the variable use_lower_ep_read should be renamed to use_device_read

Good point, I updated it in latest revision

**Comment by shijin-aws** [comment]
I will backport the first 1 commit (efa_rdm_msg one) to v1.18.x

**Pull request merged**